File: deleteUser/handler.py
import os
import json
import boto3
from botocore.exceptions import BotoCoreError, ClientError


def get_dynamodb_resource():
    if os.getenv('IS_OFFLINE', False):
        boto3.Session(
            aws_access_key_id='MockAccessKeyId',
            aws_secret_access_key='MockSecretAccessKey'
        )
        return boto3.resource('dynamodb', endpoint_url='http://localhost:8000')
    else:
        return boto3.resource('dynamodb')


def delete_user(event, context):
    client = get_dynamodb_resource()
    table = client.Table('usersTable')

    try:
        user_id = event['pathParameters']['id']
        result = table.delete_item(Key={'pk': user_id})
    except (BotoCoreError, ClientError) as e:
        # Return error message in case of exceptions
        return {
            'statusCode': 500,
            'body': json.dumps({'error_message': str(e)})
        }

    body = json.dumps({'message': f"user {user_id} deleted"})

    return {
        'status_code': result['ResponseMetadata']['HTTPStatusCode'],
        'body': body
    }

File: updateUser/handler.js
const aws = require("aws-sdk")
const { randomUUID } = require("crypto")
const settings = process.env.IS_OFFLINE
    ? {
        region: 'localhost',
        endpoint: 'http://localhost:8000',
        credentials: {
            accessKeyId: 'MockAccessKeyId',
            secretAccessKey: 'MockSecretAccessKey',
        }
    }
    : {};
const dynamodb = new aws.DynamoDB.DocumentClient(settings)

const validInputs = (body) => {
    return !(!body || !body.name || !body.phone || !body.age);
};

const createParams = (body, userId) => ({
    TableName: 'usersTable',
    Key: { pk: userId },
    UpdateExpression: "SET #name = :name, #phone = :phone, #age = :age",
    ExpressionAttributeNames: {
        "#name": "name",
        "#phone": "phone",
        "#age": "age"
    },
    ExpressionAttributeValues: {
        ":name": body.name,
        ":phone": body.phone,
        ":age": body.age,
    },
    ReturnValues: "ALL_NEW"
});

const updateUser = async (event, context) => {
    let id = event.pathParameters.id
    let body = JSON.parse(event.body)
    if (!validInputs(body)) {
        return {
            statusCode: 400,
            body: JSON.stringify({ message: 'Invalid inputs' }),
        };
    }
    const params = createParams(body, id);
    try {
        const res = await dynamodb.update(params).promise();
        return {
            statusCode: 200,
            body: JSON.stringify({ data: res.Attributes }),
        };
    } catch (error) {
        console.error('Error occurred while updating data:', error);
        return {
            statusCode: 500,
            body: JSON.stringify({ message: 'error while updating user' }),
        };
    }
}

module.exports = {
    updateUser
}